infra/aws/terraform/prow-build-cluster/kubernetes.tf

```diff
@@ -15,6 +15,8 @@ limitations under the License.
 */
 
 module "cluster_autoscaler" {
+  count = var.assume_role ? 1 : 0
```

Comment:
With changes done in `main.tf`, I don't think that we actually need this any longer. The user who created the cluster will have admin access to the cluster, so they'll be able to apply all the manifests, even when assume role is not configured.

Reply:
I tried that and it failed. It seems that terraform requires valid kubeconfig at the plan time :/ (https://github.com/hashicorp/terraform-provider-kubernetes/issues/1775#issuecomment-1185725432)